pkg/xds/bootstrap/generator.go

```diff
@@ -124,6 +127,14 @@ func (b *bootstrapGenerator) Generate(ctx context.Context, request types.Bootstr
 			params.AdminPort = b.defaultAdminPort
 		}
 	}
+	switch request.XdsConfigMode {
+	case types.DELTA:
+		params.UseDelta = true
```

Comment:
If the CP disabled `UseDeltaXds` and the DP request DeltaXDSMode, should we return the DP an error?

Reply:
No, we should allow individual DPPs to run with Delta. This approach enables a progressive migration. UseDeltaXds enforces Delta mode for all DPPs but still allows individual DPPs to revert to the previous mode if needed.

Reply:
OK, I understand it right now. With the file `pkg/xds/server/v3/components.go` we enable the deltaServer by default. I think maybe we shold rename the variable `Experimental.UseDeltaXds` because we may thought it was a swicth to enable or disable the whole `delta XDS` feature.

So we got these configurations:
1. run the delta server by default
2. CP `Experimental.UseDeltaXds` force enable deltaXDS for all DPPs
3. DP `ENV KUMA_EXPERIMENTAL_USE_DELTA_XDS` to choose use stow or delta

Reply:
I think the DPP environment variable is clear, and it shouldn't be a boolean. If it were, we wouldn’t be able to distinguish between using SOTW and "Not Defined." For the control plane configuration, you're right—I need to find a better name for it

Reply:
The docs talk about XDSTransportProtocolVariant: https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#variants-of-the-xds-transport-protocol

Reply:
Or Just APIType:

```

// APIs may be fetched via either REST or gRPC.
type ApiConfigSource_ApiType int32

const (
	// Ideally this would be 'reserved 0' but one can't reserve the default
	// value. Instead we throw an exception if this is ever used.
	//
	// Deprecated: Marked as deprecated in envoy/config/core/v3/config_source.proto.
	ApiConfigSource_DEPRECATED_AND_UNAVAILABLE_DO_NOT_USE ApiConfigSource_ApiType = 0
	// REST-JSON v2 API. The `canonical JSON encoding
	// <https://developers.google.com/protocol-buffers/docs/proto3#json>`_ for
	// the v2 protos is used.
	ApiConfigSource_REST ApiConfigSource_ApiType = 1
	// SotW gRPC service.
	ApiConfigSource_GRPC ApiConfigSource_ApiType = 2
	// Using the delta xDS gRPC service, i.e. DeltaDiscovery{Request,Response}
	// rather than Discovery{Request,Response}. Rather than sending Envoy the entire state
	// with every update, the xDS server only sends what has changed since the last update.
	ApiConfigSource_DELTA_GRPC ApiConfigSource_ApiType = 3
	// SotW xDS gRPC with ADS. All resources which resolve to this configuration source will be
	// multiplexed on a single connection to an ADS endpoint.
	// [#not-implemented-hide:]
	ApiConfigSource_AGGREGATED_GRPC ApiConfigSource_ApiType = 5
	// Delta xDS gRPC with ADS. All resources which resolve to this configuration source will be
	// multiplexed on a single connection to an ADS endpoint.
	// [#not-implemented-hide:]
	ApiConfigSource_AGGREGATED_DELTA_GRPC ApiConfigSource_ApiType = 6
)
```
In which I'd use the values: `DELTA_GRPC` and `GRPC` btw how come we're not using `AGGREGATED_GRPC` ?